Add Login page and endpoint to log in with email

njhuey opened the pull request:
Preliminary Login Page

- Create a login page with a input field for email address and a button to continue
- If the email has a uci.edu domain, redirect to the SAML login endpoint
- Otherwise, we will handle the authentication separately

njhuey committed:
implemented email verification

njhuey committed:
added login for uci users

njhuey committed:
added preliminary styling

taesungh reviewed (changes requested):
Thanks for starting this and managing to follow along with the existing file structure. Few small changes to address on handling the submission.
Could you also check your formatter settings? We want JavaScript/TypeScript files formatted with Prettier using the configuration file in the repository.

njhuey committed:
implemented small changes:

taesungh committed:
Create user login endpoint for authentication

- Check if email is valid using pydantic[email]
- Redirect UCI users to the SAML login endpoint
- TODO: implement authentication for others

taesungh committed:
Configure rewrite to proxy API in development

- Create development configuration with rewrite to serve FastAPI routes

taesungh committed:
Rebuild LoginForm with native action & validation

- Use `Form.Group` to associate label with input element
- Validate form with browser upon submission
- Use form action to send post request to login endpoint
- Add email invalid feedback message

taesungh committed:
Add UCI SSO email login description

- Include message about SSO for UCI students on login form

taesungh committed:
Refactor form validation into `ValidatingForm`

- Create `ValidatingForm` component to check form validity only upon
  attempting submission and display validation feedback as needed
